src/toast/observation_view.py

```diff
@@ -137,22 +143,32 @@ def __init__(self, obj):
     # Mapping methods
 
     def __getitem__(self, key):
-        view_name = key
-        if view_name is None:
+        if key is None:
             # Make sure the fake internal intervals are created
             trigger = self.obj.intervals[None]
             view_name = self.obj.intervals.all_name
+            inverse = False
+        else:
+            if key.startswith("~"):
+                key = key[1:]
+                inverse = True
+            else:
+                inverse = False
+            view_name = key
         if view_name not in self.obj._views:
             # View does not yet exist, create it.
             if key is not None and key not in self.obj.intervals:
                 raise KeyError(
-                    "Observation does not have interval list named '{}'".format(key)
+                    f"Observation does not have interval list named '{key}'"
                 )
             self.obj._views[view_name] = View(self.obj, key)
             # Register deleter callback
             if key is not None:
                 self.obj.intervals.register_delete_callback(key, self.__delitem__)
-        return self.obj._views[view_name]
+        if inverse:
+            return ~self.obj._views[view_name]
+        else:
+            return self.obj._views[view_name]
```

Comment:
Inverting intervals on the fly is a neat feature.  However, this is not free.  My concern is that if calling code repeatedly loops over negated intervals, then that will create a temporary object on every call.  It would be much better to just create such an interval list once.  For the use case in this PR (flagging negated intervals) it is fine, but we should avoid ever using the `~` syntax when passing views to pointing operators and mapmaking templates for example.  Another option would be to only create these negated intervals temporarily in the FlagIntervals operator and then delete them.  I am fine leaving this as-is, but we should perhaps add a warning to the documentation.

Reply:
Good point. Added a warning that is triggered when the same view is inverted more than 100 times.